_blog-src/_posts/2016-04-08-data-encryption.adoc

```diff
@@ -0,0 +1,142 @@
+---
+layout: post
+title: "Covering Your Assets: Data Encryption in API Management"
+date: 2016-04-06 08:13:00 EST
+author: rachel_yordan
+categories: security gateway
+---
+:toc: macro
+:toc-title:
+
+In a world where APIs are quickly becoming the standard, most of us understand the importance of following best practices for API security. We authenticate, authorize and throttle requests. We encrypt the data that we share with other applications (hopefully!). But we often neglect one of the most essential components of the API layer: data storage.
+
+////
+Use this pattern to avoid asciidoctor rendering the HTML comment.
+<!--more-->
+////
+
+toc::[]
+
+== Encrypting Data as a Component of API Management Security
+
+We often forget that APIs are not meant to protect data, rather, to share data across applications. While providers are certainly responsible for designing their APIs to be secure, we must be equally cautious about _each_ component of the API layer.
+
+=== Data Storage in apiman
+
+Designing, building, and maintaining an API can be a lot of work, often making security an afterthought. Using an API management platform can save you valuable time, allowing you to focus on other critical aspects of your business.
+
+Apiman was designed with the goal of providing a secure, customizable platform for API providers that is easy to use. As an open source platform, it is highly extendable and maintained by a community of developers with the common goal of streamlining API management without compromising security.
+
+*All sensitive data stored by apiman is fully encrypted* so that it is not stored in plain text, then later decrypted again on runtime or when publishing services. By default, the database storage is the H2 relational database, which uses Elasticsearch. One of the many examples of data that gets stored are the credentials that must be provided when the manager wants to publish an API and makes an authenticated REST call to the gateway.
+
+To provide an additional layer of protection we also leverage an object relational mapper built with JPA (Java Persistence API), the specification for relational data in Java-based applications. The framework that implements this specification is called Hibernate, an established and popular ORM in the Java community.
+
+Data for the API gateway and API manager are stored separately to keep components fully decoupled.
+
+For a more detailed overview, please see our http://www.apiman.io/latest/production-guide.html#_architecture_summary[API layer architecture]. Let’s swan dive into some code!
+
+=== IDataEncrypter Interface and Data Encryption Classes
+
+The interface that is used for all policies is called `IDataEncrypter`. This interface provides the structure for the classes that encrypt keys and data, which is done synchronously using AES.
```

Comment:
This should start:  "The component interface used by apiman to encrypt data (both in the API Manager and in the API Gateway) is called `IDataEncrypter`.